Adding a solution of Min Manipulations to make Strings Anagram

ultimatecoder opened the pull request:
Problem link: https://practice.geeksforgeeks.org/problems/min-manipulations-to-make-strings-anagram/0

ultimatecoder committed:
Adding a solution of Min Manipulations to make Strings Anagram

Problem link: https://practice.geeksforgeeks.org/problems/min-manipulations-to-make-strings-anagram/0

okayzed commented on programs/strings/mini_manipulations_to_make_strings_anagram.py:
```diff
@@ -0,0 +1,92 @@
+        character_stats[character] -= 1
+        if character_stats[character] == 0:
+            del character_stats[character]
+    return len(character_stats) // 2
```
shouldn't this be the sum of remaining characters? I don't understand why you use len(character_stats)

ultimatecoder replied:
Good catch :smile: ! I have added a test case for this inputs and my latest improvement in code is passing it. Could you verify this again?

ultimatecoder committed:
Fixing bug

Fixing a bug in which it was giving a wrong answer when there are
repetative characters present in one string and another string has total
different charactrs.

For example one string is "aaacb" and another is "cbdef". This should
return 3 because we can either replace "aaa" of first string with "def"
of second string or "def" of second string with "aaa" of first.

Converting hash map to a Bin class, improves a readability of a class.

ultimatecoder commented:
@okayzed I have tried to improved a readability of this problem by making a "Bin" class. In my understanding we can solve this problem by below algorithm.
* Take a bin
* Add characters of one string to bin.
* Remove characters of second string from bin.
  * If character of second string is not present in bin, that means they are different and they had to replace with characters from first string to make both as an anagram of each other.

Please verify this problem once again to check my decision of introducing `Bin` for improving readability. Thanks!

ultimatecoder committed:
Improving soution

Considering Bin as a data structure to count number of characters
present in a string. A more elegant way to solve this problem is to
calculate remaining characters in bin rather than going for some
approach externally.

By silently ignoring a character which is intended to remove from a bin,
but it is not present in bin, I can reduce external code for catching
this exception at client. I am sure this can be a bad practice, but it
looks like it is working here.

ultimatecoder committed:
Improving code

Managing length when element is removed or added will result in counting
number of chracters present in a bin in O(1). Previous attempt was
counting a character in O(N).
